### ISSUE TITLE
Train Camera-Control model

### ISSUE THREAD

**Issue by samuruph** [aigc-apps/VideoX-Fun#240]
I am trying to train a camera control model, but it seems that using the default bash file, it already load the entire fine-tuned model `models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control-Camera`. How did you train from scratch the model simply adding the trainable modules on top of Wan2.1? Should we load the original Wan2.1 model ?
Thank you!

**Comment by bubbliiiing** [comment]
Please use this code:
```python
#!/usr/bin/env python
# Convert weight from
# models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control
# to  
# models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control-Camera
import torch
from safetensors.torch import load_file, safe_open

state_dict = load_file("models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control/diffusion_pytorch_model.safetensors")

tmp_state_dict = {} 
for key in state_dict:
    if 'patch_embedding.weight' == key:
        weight = state_dict['patch_embedding.weight'][:, :16]
        weight_2 = state_dict['patch_embedding.weight'][:, 32:]
        tmp_state_dict[key] = torch.concat([weight, weight_2], 1).contiguous()
        print(key)
    else:
        tmp_state_dict[key] = state_dict[key]
state_dict = tmp_state_dict

metadata = None
from safetensors.torch import save_file
save_file(state_dict, "models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control-Camera/diffusion_pytorch_model.safetensors", metadata)
```

**Comment by samuruph** [comment]
Thank you! 
My question was more what if I want to train a camera controllable model starting from base Wan2.1 model (simply adding the new modules that you introduced in [here](https://github.com/aigc-apps/VideoX-Fun/blob/541f789188b551bde667b7f201b131b8258fd9c0/videox_fun/models/wan_transformer3d.py#L695). Because if I directly load the *Camera-Control model to train it, it seems that all the modules are already loaded pretrained.
Did I get it wrong?
Thank you very much!!

**Comment by bubbliiiing** [comment]
The initial weights of the *Camera-Control models are derived from the *Control models, and you can use the code above to perform the conversion.

We do not recommend starting training from the original Wan2.1-1.3B model, as it does not include reference image information. It is better to start from an I2V model at a minimum. Otherwise, more parameters will need to be trained — such as those for CLIP information injection — which makes training both time-consuming and difficult to converge.

**Comment by samuruph** [comment]
Okay great thank you!
One last question, I saw that the in_dim for Camera Adapter is set to 24. Basically that pretrain version doesn't use any plucker embeddings right? If one want to use need to overwrite that module right?

**Comment by bubbliiiing** [comment]
During model loading, missing parameters will be randomly initialized, and the Camera Adapter can also be initialized with random parameters.

**Comment by fushi219** [comment]
I wonder if i could just use the train_control.sh to train camera control as long as I change the "train_mode" to "control_camera_ref" and corresponding training json?

**Comment by lang25150** [comment]
> The initial weights of the *Camera-Control models are derived from the *Control models, and you can use the code above to perform the conversion.*Camera-Control 模型的初始权重源自 *Control 模型，您可以使用上面的代码执行转换。
> 
> We do not recommend starting training from the original Wan2.1-1.3B model, as it does not include reference image information. It is better to start from an I2V model at a minimum. Otherwise, more parameters will need to be trained — such as those for CLIP information injection — which makes training both time-consuming and difficult to converge.我们不建议从原始的 Wan2.1-1.3B 模型开始训练，因为它不包含参考图像信息。最好至少从 I2V 模型开始。否则，需要训练更多参数（例如用于 CLIP 信息注入的参数），这会使训练既耗时又难以收敛。

How many steps have you trained from Control model to control-camera model?  Are all of the params update with 2e-5 learning rate  including the basemodel's params ?

**Comment by fushi219** [comment]
@samuruph @lang25150  I wonder if you successfully run the training code? I have some problems trainging camera control with re10k dataset.
The learning rates are all 2e-5.

**Comment by samuruph** [comment]
I created/adapted this bash file (I have extracted clips and camera poses from nuPlan-mini, as done in [RealEstate10K](https://google.github.io/realestate10k/download.html):
```
export MODEL_NAME="models/Diffusion_Transformer/Wan2.1-Fun-V1.1-1.3B-Control-Camera"
export DATASET_NAME=""
export DATASET_META_NAME="datasets/nuplan/metadata_mini_clips.json"
# NCCL_IB_DISABLE=1 and NCCL_P2P_DISABLE=1 are used in multi nodes without RDMA. 
# export NCCL_IB_DISABLE=1
# export NCCL_P2P_DISABLE=1
NCCL_DEBUG=INFO

# CUDA_VISIBLE_DEVICES=0 
# accelerate launch --mixed_precision="bf16" scripts/wan2.1_fun/train_singleview_control.py \
accelerate launch --zero_stage 3 --zero3_save_16bit_model true --zero3_init_flag true --use_deepspeed --deepspeed_config_file config/zero_stage3_config.json --deepspeed_multinode_launcher standard scripts/graphdreamerwan2.1_fun/train_singleview_control_demo.py \
  --base_config_path="config/models/wan2.1/wan_civitai.yaml" \
  --additional_config_path="config/models/wan2.1/singleviewgraphdreamerwan_civitai.yaml" \
  --pretrained_model_name_or_path=$MODEL_NAME \
  --train_data_dir=$DATASET_NAME \
  --train_data_meta=$DATASET_META_NAME \
  --image_sample_size=1024 \
  --video_sample_size=256 \
  --token_sample_size=512 \
  --video_sample_stride=1 \
  --video_sample_n_frames=80 \
  --train_batch_size=1 \
  --video_repeat=1 \
  --gradient_accumulation_steps=1 \
  --dataloader_num_workers=12 \
  --num_train_epochs=100 \
  --checkpointing_steps=500 \
  --learning_rate=2e-05 \
  --lr_scheduler="constant_with_warmup" \
  --lr_warmup_steps=100 \
  --seed=42 \
  --validation_prompts "The video is captured from a camera mounted on a car, facing forward center" \
  --validation_paths "datasets/nuplan/mini/2021.05.12.22.00.38_veh-35_01008_01518_CAM_F0.mp4" \
  --output_dir="output_dir" \
  --gradient_checkpointing \
  --mixed_precision="bf16" \
  --adam_weight_decay=3e-2 \
  --adam_epsilon=1e-10 \
  --vae_mini_batch=1 \
  --max_grad_norm=0.05 \
  --random_hw_adapt \
  --training_with_video_token_length \
  --enable_bucket \
  --video_length_drop_start=0.0 \
  --video_length_drop_end=1.0 \
  --uniform_sampling \
  --low_vram \
  --train_mode="control_camera_ref" \
  --control_ref_image="first_frame" \
  --trainable_modules "." \
  --n_views 1 \
  --sanity_check \
  --validation_steps 10000000 \
  --validation_epochs 10000000 \
  --tracker_project_name="image2video-wan2.1fun-cameracontrol-fine-tune-nuplan"
```
I simply refactor some configs and so on. I trained for few epochs, so the results are not amazing, but I guess if you train longer or with more compute results will generate pretty nicely. For now, it follows the camera input I give a bit, but on such complex automotive scenarios fails in agents generation consistency. I simply added some args (not important), so it might not work directly on `train_control.py` script.

**Comment by EmmaThompson123** [comment]
@bubbliiiing Hello, what is the training data of Wan2.1-Fun-V1.1-*B-Control-Camera, is it just [RealEstate10K](https://google.github.io/realestate10k/download.html) ?
